students/astrawmyer/lesson03/mailroom.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+donors = [["Manny Machado",[12.2,2.51,3.20]],["Adam Jones",[1024.14,22.21,323.45]],["Chris Davis",[3.2,5.55,4.20]]]
+
+
+def displaylist():    
+    for sublist in donors:
+        print(sublist[0])
+
+def writeletter(name,amount):
+    print('Dear {},'.format(name))
+    print("Thank you for donating ${:.2f} to the Human Fund. Your money will be used appropriately.".format(amount))
+
+def writereport(donors):
+    print('Donor Name                | Total Given | Num Gifts | Average Gift')
+    print('-'*67)
+    for i in donors:
+        print('{1:27}${0:11.2f}{2:12}  ${3:12.2f}'.format(*i))
+
+
+# Function to send a thank you letter.
+def thankyou():
+    loop_trigger = True
+    while loop_trigger == True:
+        input_name = input("Enter full name: ")
+        for i, donor_row in enumerate(donors):
+            if donor_row[0] == input_name:
+                donation = float(input("Enter donation amount:"))
+                donors[i][1].append(float(donation))
+                print(donors)
+                writeletter(donors[i][0],donors[i][1][len(donors[i][1])-1])
+                loop_trigger = False
+                break
+            elif input_name == 'list':
+                displaylist()
+                break
```

Comment:
missing requirement: 
"If the user types a name not in the list, add that name to the data structure and use it."